stdm_config/views.py

```diff
@@ -217,68 +218,59 @@ def EntityDetailView(request, profile_name, short_name):
-		has_spatial_column = True
-	
-	if social_tenure.is_str_entity(entity):
-		is_str_entity = True
+	errors = None
+	items = None
+	entity_name = None
+	columns = []
+	lookup_summaries = {}
+	spatial_results = None
 
-	with connection.cursor() as cursor:
+	entity = prof.entity(entity_short_name)
+	try:
+		if not entity:
+			raise Exception(
+				"{} entity was not found in {} profile".format(entity_short_name, profile_name)
+			)
+		social_tenure = prof.social_tenure       
+		entity_name = entity.name
+
+		if entity.has_geometry_column():
+			has_spatial_column = True
 		
-		query = fetch_entity_records(prof, entity)
-		cursor.execute(query)
-		data1 = cursor.fetchall()
-		items = [zip([key[0] for key in cursor.description], row) for row in data1]
+		if social_tenure.is_str_entity(entity):
+			is_str_entity = True
 		
-		for key in cursor.description:
-			columns.append(toHeader(key[0]))
-
-	lookup_summaries = EntityLookupSummaries(prof, entity)
-
-	spatial_results = None
-	if has_spatial_column:
-		spatial_columns = entity.geometry_columns()
-		for sp in spatial_columns:
-			spatial_column = sp.name
-			break
-		query_join_columns = entity_columns_to_query(entity)
-		query = "SELECT row_to_json(fc) \
-				FROM \
-				( SELECT 'FeatureCollection' AS TYPE, \
-						array_to_json(array_agg(f)) AS features \
-				FROM \
-					(SELECT 'Feature' AS TYPE, \
-							ST_AsGeoJSON({0}_view.{1},4326)::JSON AS geometry, \
-							row_to_json( (SELECT p FROM ( SELECT {2}) AS p)) AS properties \
-					FROM {3}_view) AS f) AS fc;	".format(entity_name, spatial_column, ','.join(query_join_columns), entity_name)
 		with connection.cursor() as cursor:
+			query = fetch_entity_records(entity)
+			LOGGER.info("%s entity details full query %s", entity_short_name, query)
 			cursor.execute(query)
-			spatial_result = cursor.fetchone()
-			map_data = spatial_result[0]
-			spatial_results = json.dumps(map_data)
-	return render(request,'dashboard/entity.html', {'entity':entity,'profile':profile_name,'entity_name':entity_name,'data':items,'columns':columns,'has_spatial_column':has_spatial_column,'is_str_entity':is_str_entity,'lookup_summaries':lookup_summaries, 'spatial_result':spatial_results })
-
-@csrf_exempt
-def fetch_spatial_data(request, profile_name, entity_short_name):
-	config = GetConfig("Web")
-	if config is None or not config.complete:
-		return render(request, 'dashboard/no_config.html',)
-	stdm_config = GetStdmConfig("Web")
-	prof = stdm_config.profile(profile_name)
-	entity = prof.entity(entity_short_name)
-	if not entity.has_geometry_column():
-		return None
+			data1 = cursor.fetchall()
+			items = [zip([key[0] for key in cursor.description], row) for row in data1]
+			
+			for key in cursor.description:
+				columns.append(toHeader(key[0]))
+		
+		lookup_summaries = EntityLookupSummaries(prof, entity)
+		spatial_results = None
+		if has_spatial_column:
+			spatial_results = entity_geojson(entity)
+	except Exception as e:
+		print(e.args)
```

Comment:
Should this print be there or it was for debugging?

Reply:
To remove, should be a LOGGER instead